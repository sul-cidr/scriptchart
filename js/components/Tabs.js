import React from "react";

import PropTypes from "prop-types";

const Tab = props => {
  const { name } = props.tab;
  const { activeTab, changeActiveTab } = props;

  return (
    <li
      className={name === activeTab ? "is-active" : ""}
      onClick={() => changeActiveTab(name)}
    >
      <a>
        <span className="icon is-small">
          <i className={props.icon} />
        </span>
        <span>{name}</span>
      </a>
    </li>
  );
};

class Tabs extends React.Component {
  static propTypes = {
    tabList: PropTypes.array.isRequired,
    activeTab: PropTypes.string,
    changeActiveTab: PropTypes.func
  };

  render() {
    return (
      <div className="tabs is-toggle is-boxed">
        <ul>
          {this.props.tabList.map(tab => (
            <Tab
              tab={tab}
              key={tab.name}
              icon={tab.icon}
              activeTab={this.props.activeTab}
              changeActiveTab={this.props.changeActiveTab}
            />
          ))}
        </ul>
      </div>
    );
  }
}

export default Tabs;
