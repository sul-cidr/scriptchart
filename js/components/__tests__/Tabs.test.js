/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from '../Tabs';


describe('Tabs test', () => {
  it('Tabs should match snapshot', () => {
    const component = renderer.create(<Tabs
      tabList={[]} activeTab={undefined} changeActiveTab={undefined} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
