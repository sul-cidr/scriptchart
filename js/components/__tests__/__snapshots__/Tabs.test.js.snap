// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs test Tabs should match snapshot 1`] = `
<div
  className="tabs is-toggle is-boxed"
>
  <ul />
</div>
`;
